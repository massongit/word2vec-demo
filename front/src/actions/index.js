import * as types from "./types"
import {createAction} from "redux-actions"

/**
 * Word2Vecによる計算結果の表示ActionのActionCreator
 * @type {function}
 */
export const showSimilarWords = createAction(types.SHOW_SIMILAR_WORDS, parameter => ({
    words: parameter.words,
    keywords: parameter.keywords
}))
