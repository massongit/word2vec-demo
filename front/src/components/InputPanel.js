import React from "react"
import PanelBody from "../containers/PanelBody"
import Input from "../containers/Input"
import {Panel, Row} from "react-bootstrap"
import {FormattedMessage} from "react-intl"

/**
 * 入力部のパネル
 * @returns {element} 入力部のパネル
 */
const InputPanel = () => (
    <Row>
        <Panel>
            <Panel.Heading>
                <FormattedMessage id="title.input"/>
            </Panel.Heading>
            <PanelBody
                messageId="description.input"
                additionalDescription={
                    <ul>
                        <li>
                            <FormattedMessage id="description.input.supplement.1"/>
                        </li>
                        <li>
                            <FormattedMessage id="description.input.supplement.2"/>
                        </li>
                    </ul>
                }
            >
                <Input/>
            </PanelBody>
        </Panel>
    </Row>
)

export default InputPanel
