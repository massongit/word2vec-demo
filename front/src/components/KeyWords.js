import React from "react"
import {FormattedMessage} from "react-intl"
import PropTypes from "prop-types"

/**
 * positiveとnegativeのPropTypes
 * @type {{positive: shim, negative: shim}}
 */
export const propTypesPN = {
    positive: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    negative: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
}

/**
 * キーワード
 */
class KeyWords extends React.Component {
    static propTypes = propTypesPN

    render() {
        return (
            <p>
                <label htmlFor="keywords">
                    <FormattedMessage id="header.keywords"/>
                </label>
                <p id="keywords">
                    {
                        (() => {
                            let r = []

                            for (const v of [["+", this.props.positive], ["-", this.props.negative]]) {
                                if (v[1]) {
                                    r = r.concat(v[1].map(w => (
                                        v[0] + w
                                    )))
                                }
                            }

                            return r.join(" ")
                        })()
                    }
                </p>
            </p>
        )
    }
}

export default KeyWords
