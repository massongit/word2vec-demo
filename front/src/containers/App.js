import App from "../components/App"
import {connect} from "react-redux"

const mapStateToProps = state => ({
    similar: state.showSimilarWords.words.similar,
    method: state.setMethod.method
})

/**
 * ルート要素のContainer
 */
export default connect(mapStateToProps)(App)
