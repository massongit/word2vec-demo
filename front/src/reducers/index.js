import loading from "./loading"
import setMethod from "./setMethod"
import setMethods from "./setMethods"
import showSimilarWords from "./showSimilarWords"
import {combineReducers} from "redux"

/**
 * ルートReducer
 */
export default combineReducers({
    showSimilarWords,
    loading,
    setMethods,
    setMethod
})
