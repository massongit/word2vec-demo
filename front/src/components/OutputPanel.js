import React from "react"
import PanelBody from "../containers/PanelBody"
import Information from "../containers/Information"
import {Panel, Row} from "react-bootstrap"
import {FormattedMessage} from "react-intl"
import KeyWords from "../containers/KeyWords"

/**
 * 計算結果表示部のパネル
 */
const OutputPanel = () => (
    <Row>
        <Panel>
            <Panel.Heading>
                <FormattedMessage id="title.result"/>
            </Panel.Heading>
            <PanelBody>
                <KeyWords/>
            </PanelBody>
            <Information/>
        </Panel>
    </Row>
)

export default OutputPanel
