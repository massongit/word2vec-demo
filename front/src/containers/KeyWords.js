import KeyWords from "../components/KeyWords"
import {connect} from "react-redux"

const mapStateToProps = state => ({
    positive: state.showSimilarWords.words.positive,
    negative: state.showSimilarWords.words.negative
})

/**
 * キーワードのContainer
 */
export default connect(mapStateToProps)(KeyWords)
