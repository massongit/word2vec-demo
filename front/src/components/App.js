import React from "react"
import PropTypes from "prop-types"
import Header from "./Header"
import InputPanel from "./InputPanel"
import MethodPanel from "./MethodPanel"
import DescriptionPanel from "../containers/DescriptionPanel"
import OutputPanel from "./OutputPanel"
import {Grid} from "react-bootstrap"

/**
 * similarのPropTypes
 * @type {{similar: shim}}
 */
export const propTypesSimilar = {
    similar: PropTypes.arrayOf(PropTypes.shape({
        word: PropTypes.string.isRequired,
        cosine: PropTypes.number.isRequired
    }).isRequired).isRequired
}

/**
 * ルート要素
 */
class App extends React.Component {
    static propTypes = {
        ...propTypesSimilar,
        method: PropTypes.string
    }

    render() {
        return (
            <Grid>
                <Header/>
                <DescriptionPanel/>
                <MethodPanel/>
                {
                    (() => {
                        if (this.props.method) {
                            return (<InputPanel/>)
                        }
                    })()
                }
                {
                    (() => {
                        // 分散表現による計算が行われていない場合には表示しない
                        if (0 < this.props.similar.length) {
                            return (<OutputPanel/>)
                        }
                    })()
                }
            </Grid>
        )
    }
}

export default App
