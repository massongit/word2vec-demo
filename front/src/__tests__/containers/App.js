import React from "react"
import App from "../../containers/App"
import OutputPanel from "../../components/OutputPanel"
import InputPanel from "../../components/InputPanel"
import rootReducer from "../../reducers"
import setMethodsState from "../../test_data/setMethodsState"
import {shallow} from "enzyme"
import {createStore} from "redux"
import {doSnapshot, functions} from "./KeyWords"
import {dispatchActions} from "../reducers"
import {setMethod, setMethods} from "../../actions"
import {setMethodParameter3} from "../../test_data"

export const functions2 = [
    undefined,
    store => {
        dispatchActions(store, setMethods(setMethodsState))
    },
    store => {
        dispatchActions(store, setMethod(setMethodParameter3))
    },
    functions
]

let store

/**
 * テストの前処理
 */
const beforeProcess = () => (
    shallow(
        <App
            store={store}
        />
    ).dive()
)

describe("containers/PanelBody", () => {
    beforeEach(() => {
        store = createStore(rootReducer)
    })

    it("Componentが正しく配置されている", () => {
        doSnapshot(functions2, store, beforeProcess)
    })

    it("子要素にInputPanelが含まれる", () => {
        for (const f of functions2) {
            if (f) {
                f(store)
            }
        }

        expect(beforeProcess().children().contains(<InputPanel/>)).toBeTruthy()
    })

    it("子要素にOutputPanelが正しく配置される", () => {
        for (const v of [[functions2[0], false], [functions2[1], false], [functions2[2], false], [functions2[3], true]]) {
            if (v[0]) {
                v[0](store)
            }

            expect(beforeProcess().children().contains(<OutputPanel/>)).toEqual(v[1])
        }
    })
})
