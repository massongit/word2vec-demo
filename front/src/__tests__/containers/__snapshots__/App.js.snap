// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/KeyWords 初期状態からStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyWords
    dispatch={[Function]}
    negative={
      Array [
        "男",
      ]
    }
    positive={
      Array [
        "王",
        "女",
      ]
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="keywords"
        >
          <FormattedMessage
            id="header.keywords"
            values={Object {}}
          />
        </label>,
        <p
          id="keywords"
        >
          +王 +女 -男
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormattedMessage
            id="header.keywords"
            values={Object {}}
          />,
          "htmlFor": "keywords",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "header.keywords",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+王 +女 -男",
          "id": "keywords",
        },
        "ref": null,
        "rendered": "+王 +女 -男",
        "type": "p",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="keywords"
          >
            <FormattedMessage
              id="header.keywords"
              values={Object {}}
            />
          </label>,
          <p
            id="keywords"
          >
            +王 +女 -男
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              id="header.keywords"
              values={Object {}}
            />,
            "htmlFor": "keywords",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "header.keywords",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+王 +女 -男",
            "id": "keywords",
          },
          "ref": null,
          "rendered": "+王 +女 -男",
          "type": "p",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`containers/PanelCardBody Componentが正しく配置されている 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    similar={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <InjectIntl(DescriptionCard) />,
        <MethodCard />,
        undefined,
        undefined,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <InjectIntl(DescriptionCard) />,
          <MethodCard />,
          undefined,
          undefined,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`containers/PanelCardBody Componentが正しく配置されている 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    similar={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <InjectIntl(DescriptionCard) />,
        <MethodCard />,
        undefined,
        undefined,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <InjectIntl(DescriptionCard) />,
          <MethodCard />,
          undefined,
          undefined,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`containers/PanelCardBody Componentが正しく配置されている 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    method="fasttext"
    similar={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <InjectIntl(DescriptionCard) />,
        <MethodCard />,
        <InputCard />,
        undefined,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <InjectIntl(DescriptionCard) />,
          <MethodCard />,
          <InputCard />,
          undefined,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`containers/PanelCardBody Componentが正しく配置されている 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    method="fasttext"
    similar={
      Array [
        Object {
          "cosine": 0.6129051446914673,
          "word": "女王",
        },
        Object {
          "cosine": 0.5734513998031616,
          "word": "大王",
        },
        Object {
          "cosine": 0.5623782873153687,
          "word": "王女",
        },
        Object {
          "cosine": 0.557570219039917,
          "word": "王妃",
        },
        Object {
          "cosine": 0.536829948425293,
          "word": "デュエルモンスターズビギナーズパック",
        },
        Object {
          "cosine": 0.5333640575408936,
          "word": "デュエルモンスターズエキスパート",
        },
        Object {
          "cosine": 0.5321550369262695,
          "word": "戯",
        },
        Object {
          "cosine": 0.5298371911048889,
          "word": "遊戯",
        },
        Object {
          "cosine": 0.5250808000564575,
          "word": "デュエルモンスターズ",
        },
        Object {
          "cosine": 0.525033175945282,
          "word": "王家",
        },
      ]
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <InjectIntl(DescriptionCard) />,
        <MethodCard />,
        <InputCard />,
        <OutputCard />,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <InjectIntl(DescriptionCard) />,
          <MethodCard />,
          <InputCard />,
          <OutputCard />,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
